rzup/src/cli/commands.rs

```diff
@@ -0,0 +1,300 @@
+    #[arg(value_parser=component_parser())]
+    name: String,
+    /// Version of component to activate
+    version: String,
+}
+
+impl UseCommand {
+    pub(crate) fn execute(self, rzup: &mut Rzup) -> Result<()> {
+        let version = Version::parse(&self.version)
+            .map_err(|_| RzupError::InvalidVersion(self.version.clone()))?;
+
+        if rzup.version_exists(&self.name, &version)? {
+            rzup.set_active_version(&self.name, version.clone())?;
+            println!(
+                "Successfully set {} version {} as active",
+                self.name, version
+            );
+        } else {
+            println!(
+                "! Version {} of {} is not installed.\n  Please use 'rzup install {} {}' to install",
+                version, self.name, self.name, version
+            );
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Parser)]
+pub(crate) struct CheckCommand;
+
+impl CheckCommand {
+    pub(crate) fn execute(&self, rzup: &Rzup) -> Result<()> {
+        let components = rzup.registry.list_components();
+        if components.is_empty() {
+            println!("! Nothing is installed\n  Please use 'rzup install' to install.");
+            return Ok(());
+        }
+
+        let mut results = Vec::new();
+
+        for component in components {
+            let id = component.id();
+            rzup.emit(RzupEvent::CheckUpdates {
+                id: Some(id.to_string()),
+            });
+
+            let latest_version = rzup.get_latest_version(id)?;
+            let installed_versions = rzup.list_versions(id)?;
+
+            if let Some(max_installed) = installed_versions.iter().max() {
+                if !rzup.version_exists(id, &latest_version)? {
+                    results.push(format!(
+                        "{} - {} : {} -> {}",
+                        id.bold(),
+                        "Update Available".bold().yellow(),
+                        max_installed,
+                        latest_version,
+                    ));
+                } else {
+                    results.push(format!(
+                        "{} - {} : {}",
+                        id.bold(),
+                        "Up to date".bold().green(),
+                        max_installed
+                    ));
+                }
+            }
+        }
+
+        rzup.emit(RzupEvent::CheckUpdates { id: None });
+
+        for result in results {
+            println!("{}", result);
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Parser)]
+pub(crate) struct UninstallCommand {
+    /// Name of component to uninstall
+    #[arg(value_parser=component_parser())]
+    name: String,
+    /// Version of the component to uninstall
+    version: String,
+}
+
+impl UninstallCommand {
+    pub(crate) fn execute(&self, rzup: &mut Rzup) -> Result<()> {
+        let version = Version::parse(&self.version)
+            .map_err(|_| RzupError::InvalidVersion(self.version.clone()))?;
```

Comment:
could this maybe instead an implementation of `From<semver::Version> for RzupError`?